lucene/core/src/java/org/apache/lucene/util/compress/LZ4.java

```diff
@@ -53,6 +53,27 @@ private LZ4() {}
    */
   public static final int MAX_DISTANCE = 1 << 16; // maximum distance of a reference
 
+  /**
+   * Longer lookback window size. Cf. {@link #MAX_DISTANCE}. This allows the context window to be
+   * 256k instead of the default 64k, and can provide substantial compression ratio and performance
+   * boost for data where the repetition period is longer.
+   */
+  public static final int EXTENDED_MAX_DISTANCE = (1 << 18) - 1;
+
+  /**
+   * There are some use cases (e.g., 256k block-level compression applied over index files) where
+   * the period of pattern repetition is longer. Such cases benefit from a combination of {@link
+   * HighCompressionHashTable} and a longer lookback window ({@link #EXTENDED_MAX_DISTANCE} instead
+   * of {@link #MAX_DISTANCE}). The benefits are both in compression (real-world cases with ~3x
+   * improved compression!), but also in latency/CPU-efficiency, in some cases with &gt;2x faster
+   * execution.
+   *
+   * <p>We want to support lz4 with {@link #EXTENDED_MAX_DISTANCE} for these special cases, but also
+   * provide {@link #DEFAULT_EXTENDED_MAX_DISTANCE} to allow tests to be run exercising lz4 with
+   * {@link #EXTENDED_MAX_DISTANCE}.
+   */
+  public static final boolean DEFAULT_EXTENDED_MAX_DISTANCE = false;
```

Comment:
add sys prop here?

Reply:
could do ... I opted not to because the main use for this would be to run full test suite with `EXTENDED_MAX_DISTANCE` config, and that would be a bit more involved than simply adding a sysprop. If we're going make this configurable via sysprop there would be some benefit, but IMO only if we go as far as supporting configuration via project property for running tests.

Practically this only affects the lookback window for existing "inner" lz4 -- it has no impact on the configuration of lz4 as employed by TeeDirectory/CompressingDirectory. For these, `EXTENDED_MAX_DISTANCE` defaults to `true`, and is configurable via `solrconfig.xml`.